src/main/perl/FileWriter.pm

```diff
@@ -215,21 +231,43 @@ sub close
         *$self->{save} = 0;
         my $content_ref = *$self->{buf};
 
+        if (*$self->{original_from_source}) {
+            # Report changes compared to source
+            my $src_original_content = *$self->{original_content};
+            if (defined($src_original_content)) {
+                my $src_diff = diff(\$src_original_content, $content_ref, { STYLE => "Unified" });
+                my $src_changed = $src_diff ? 1 : 0;
+                if ($src_changed) {
+                    if(*$self->{options}->{sensitive}) {
+                        $self->verbose("Changes compared to source for $filename are not reported due to sensitive content");
+                    } else {
+                        $self->verbose("Changes compared to source for $filename:");
+                        $self->report($src_diff) if $self->is_verbose();
+                    }
+                } else {
+                    $self->verbose("No changes compared to source for $filename");
+                }
+            } else {
+                $self->verbose("No original source content for $filename");
+            }
+        }
+
         # Always read and (try to) determine the diff
-        my $content_orig;
-        if (defined(*$self->{content_orig})) {
-            $self->debug(2, "Using original contents from $filename");
-            $content_orig = *$self->{content_orig};
+        my $original_content;
+        if (defined(*$self->{original_content}) && !*$self->{original_from_source}) {
+            $self->debug(2, "Using existing original content for $filename");
+            $original_content = *$self->{original_content};
         } else {
             # missing_ok=1 mimics original LC::Check::file behaviour
-            $content_orig = $self->_read_contents($filename, event => \%event, missing_ok => 1);
+            $original_content = $self->_read_contents($filename, event => \%event, missing_ok => 1);
```

Comment:
Why is this extra read needed here? The conditions under which this line is executed are not trivially fathomable (cf. comment above) so I think some extra comments here about how we can end up here and why we need this line would be useful.